src/mds/MDSMap.cc

```diff
@@ -177,11 +177,20 @@ void MDSMap::dump(Formatter *f) const
     f->close_section();
   }
   f->close_section();
-  f->open_array_section("data_pools");
-  for (set<int64_t>::const_iterator p = data_pools.begin(); p != data_pools.end(); ++p)
-    f->dump_int("pool", *p);
-  f->close_section();
-  f->dump_int("metadata_pool", metadata_pool);
+
+  if (data_pools.empty())
+    f->dump_string("data_pools", "None");
+  else {
+    f->open_array_section("data_pools");
+    for (set<int64_t>::const_iterator p = data_pools.begin(); p != data_pools.end(); ++p)
+      f->dump_int("pool", *p);
+    f->close_section();
+  }
+
+  if (metadata_pool == -1)
+    f->dump_string("metadata_pool", "None");
```

Comment:
If we could output null then this would be good, but unfortunately the Formatter class doesn't support that.  Outputting a string is not an improvement either: it makes sense to human eyes but if a piece of code is parsing the output it might be confused.